wepool-app/src/components/RideDetailsModal.tsx

```diff
@@ -1,19 +1,37 @@
 import { StyleSheet, View, Text, TouchableOpacity, Modal, TextInput } from 'react-native'
 import {Button} from "./Button"
 import { useThemeColors } from "../hooks/useThemeColors";
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { FontAwesome } from '@expo/vector-icons';
 import { Divider } from '@rneui/themed';
 
 type Props = {
     openDetails: boolean,
-    handleOnPressDetails: any,
+    rides: Ride[],
+    handleOpenDetails: any,
+    rideId: number
 }
 
-export const RideDetailsModal = ({openDetails, handleOnPressDetails}: Props) => {
+export const RideDetailsModal = ({openDetails, handleOpenDetails, rides, rideId}: Props) => {
     
     const { colors } = useThemeColors();
     const backgroundColor = colors.tint
+    
+    // const [selectedRide, setSelectedRide] = useState<Ride>();
+
+    // useEffect(() => {
+    //     setSelectedRide(rides.find(x => {
+    //         return x.id === rideId;
+    //     }));
+    //   });
+
+    const selectedRide = rides.find(x => {
```

Comment:
Nit: try to use variables with significant naming, in this case `rides.find(ride => ride.id === rideId)`